cmd/keymasterd/app.go

```diff
@@ -518,7 +519,6 @@ func (state *RuntimeState) writeFailureResponse(w http.ResponseWriter,
 	if code == http.StatusUnauthorized && returnAcceptType != "text/html" {
 		w.Header().Set("WWW-Authenticate", `Basic realm="User Credentials"`)
 	}
-	w.WriteHeader(code)
```

Comment:
why are you changing this? it is still a 401... with this change it becomes a 200.

Reply:
Because writing the error code here breaks the redirection to the OAuth2 handler.
I made sure to write the error code in the API case as well as the when not writing a login page, so I don't think this breaks anything.